python/hidet/testing/models/llama.py

```diff
@@ -450,7 +449,9 @@ def get_compiled_model(name='decapoda-research/llama-7b-hf', device='cuda', opt=
     flow_graph = build_flow_graph(model, device=device)
 
     if opt:
-        flow_graph = hidet.graph.optimize(flow_graph)
+        with hidet.graph.PassContext() as ctx:
+            ctx.reduce_cuda_compile_mem(True)
```

Comment:
We use some heristic rule to determine whether to turn on this.

For example, if the model weights used more than 1/3 gpu memory to store its weights, we can apply the transformation, otherwise disable it.

```python
def reduce_cuda_compile_mem(flag: Optional[bool] = None):
    """
    flag: Optional[bool]
        None - determined automatically
        True - always ...
        False - never ...
    """
    ...
```
We add the instrument when the `flag is None or flag is True`and do the checking in the instrument.